### PR TITLE
926 gcp cluster logs

### LINKED ISSUE #926: Cluster logs location issue with GCP MultiWorkspace Deployment
As per the latest GCP Databricks release Cluster logs location does not allow gcs bucket path or mounted location as an input for cluster logs location, it will only allow dbfs(root GCS bucket) location as an input. This will cause issues to populate the Spark related tables in a Multi workspace deployment.

### PR DESCRIPTION
**Pull request by aman-db**
closes #926

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala [hunk 1]
```diff
@@ -949,6 +949,24 @@ trait BronzeTransforms extends SparkSessionWrapper {
...
 
+  /**
+   *
+   * @param cloudProvider
+   * @param inputCol
+   * @param isMultiWorkSpaceDeployment
+   * @param organisationId
+   * @return
+   */
+  private def fetchClusterLogConfiguration(cloudProvider: String, inputCol: Column,
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:960]: reformat signature

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala [hunk 2]
```diff
@@ -949,6 +949,24 @@ trait BronzeTransforms extends SparkSessionWrapper {
...
+                                   isMultiWorkSpaceDeployment: Boolean, organisationId: String): Column = {
+    if(cloudProvider.toLowerCase() == "gcp" && isMultiWorkSpaceDeployment)
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:962]: add curly braces around if the if logic and remove the old comment lines.

Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:962]: shouldn't we only do this for GCP REMOTE workspaces? GCP local master / driving workspaces should just use the log path in the cluster config, no?

**Review by GeekSheikh — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 1 above)
Comment (sriram251-code) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:960]: what is inputCol? have some documentation regarding it.
add some documentation in the function.

**Review by sriram251-code — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 2 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:962]: done

Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:962]: yes, we are doing this only for GCP multiworkspace

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 1 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:960]: done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 2 above)
Comment (GeekSheikh) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala]: add where local workspace orgID != currentOrgId (current from config)

Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala]: done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 1 above)
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:963]: please reformat the signature

**Review by sriram251-code — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala (hunk 1 above)
Comment (aman-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala:963]: done

**Review by sriram251-code — CHANGES_REQUESTED** [review]
Please test this in AWS and Azure

**Comment by aman-db** [comment]
> Please test this in AWS and Azure

@sriram251-code Done. Please check

**Review by sriram251-code — APPROVED** [review]

**Review by gueniai — APPROVED** [review]
lgtm!

**Pull request merged**